### DISCUSSION TITLE
.Debug() method returns incorrect query

### DISCUSSION THREAD

**Discussion by banchmaxedgedenergy** [Gremlinq/ExRam.Gremlinq#708] [General]
Hello everyone,

Here is the simple query I built:
` var result =_g.V('someId').Out<Makes>().OfType<ProductModel>().Debug();`

The result is:
`g.V(_a).out(_b).hasLabel(_c)`

I tried to debug, and figured that internally when Debug() method is executed, the serializer returns an object with Bindings collection where all those _a and _b and _c are located:
![image](https://user-images.githubusercontent.com/101670266/180052847-6191918a-9a04-4011-aa24-732781e0f135.png)

Also I look at what is used in Environment.Debugger and figured that default debugger (which is GroovyGremlinQueryDebugger) supports only BytecodeGremlinQuery types, when TryToString() method is being called it expects to receive BytecodeGremlinQuery, however in my case the type is  GroovyGremlinQuery:
![image](https://user-images.githubusercontent.com/101670266/180053735-bde1544b-0905-4ca3-a4e9-196aa8eedf2a.png)

And that's why the result is not full.

Maybe my environment is setup incorrectly, or this part is not implemented yet...

Please advise.
Thank you!

P.S. My problem is I'd like to see the real query that is being sent to the server.
Thanks!

**Comment by danielcweber** [comment]
Seems like an omission. Handling a GroovyGremlinQuery there is easy. Feeling brave enough to craft a pull request?

**Reply by banchmaxedgedenergy** [reply]
Haha, I would like to do it, but it seems I don't get how to process GroovyGremlinQuery, it seem all infrastructure is there, so maybe if you point me in a right direction I will create pr. I tried to do it, the only way I was able to find is to change implementation of private method .Debug() in GremlinQuery class, so that serializer creates  BytecodeGremlinQuery instead of GroovyGremlinQuery, but i'm not sure it's a correct way. Thanks!
